src/web3/Farm.ts

```diff
@@ -1,656 +1,775 @@
-    )
-    const tokenAccountBKey = await getAssociatedTokenAddress(
-        mintB, 
-        userPublicKey,
-        null,
-        userTargetTokenType === 'spl-token-2022' ? TOKEN_2022_PROGRAM_ID : TOKEN_PROGRAM_ID
-    )
-
-    const accountObj = {
-        owner: userPublicKey,
-        authority: authorityKey,
-        poolState: poolIdKey,
-        userPoolLiquidity: liquidityAccountKey,
-        token0Account: tokenAccountAKey,
-        token1Account: tokenAccountBKey,
-        token0Vault: poolVaultKeyA,
-        token1Vault: poolVaultKeyB,
-        tokenProgram: TOKEN_PROGRAM_ID,
-        tokenProgram2022: TOKEN_2022_PROGRAM_ID,
-        vault0Mint: mintA,
-        vault1Mint: mintB
-    }
-
-    if (!isDeposit) accountObj["memoProgram"] = MEMO_ID
-    return accountObj
+const getAccountsForSwappingTokens = async (
+  mintA: GAMMAToken,
+  mintB: GAMMAToken,
+  userSourceTokenType: 'spl-token' | 'native' | 'spl-token-2022' | '',
+  userTargetTokenType: 'spl-token' | 'native' | 'spl-token-2022' | '',
+  poolState: any,
+  userPublicKey: PublicKey
+) => {
+  const configIdKey = await getAmmConfigId(0)
+  const mintAPublicKey = new PublicKey(mintA?.address)
+  const mintBPublickey = new PublicKey(mintB?.address)
+  const poolIdKey = await getPoolIdKey(configIdKey, mintAPublicKey, mintBPublickey)
+  const authorityKey = await getAuthorityKey()
+
+  const inputTokenAccount = await getAssociatedTokenAddress(
+    mintAPublicKey,
+    userPublicKey,
+    null,
+    userSourceTokenType === 'spl-token-2022' ? TOKEN_2022_PROGRAM_ID : TOKEN_PROGRAM_ID
+  )
+
+  const outputTokenAccount = await getAssociatedTokenAddress(
+    mintBPublickey,
+    userPublicKey,
+    null,
+    userTargetTokenType === 'spl-token-2022' ? TOKEN_2022_PROGRAM_ID : TOKEN_PROGRAM_ID
+  )
+
+  return {
+    ammConfig: configIdKey,
+    poolState: poolIdKey,
+    inputVault: poolState.token0Vault,
+    outputVault: poolState.token1Vault,
+    observationState: poolState.observationKey,
+    payer: userPublicKey,
+    inputTokenAccount: inputTokenAccount,
+    outputTokenAccount: outputTokenAccount,
+    inputTokenMint: mintAPublicKey,
+    outputTokenMint: mintBPublickey,
+    inputTokenProgram: userSourceTokenType === 'spl-token-2022' ? TOKEN_2022_PROGRAM_ID : TOKEN_PROGRAM_ID,
+    outputTokenProgram: userTargetTokenType === 'spl-token-2022' ? TOKEN_2022_PROGRAM_ID : TOKEN_PROGRAM_ID,
+    authority: authorityKey
+  }
 }
 
 const handleSlippageCalculation = (amount: string, slippage: number, isDeposit: boolean): string => {
-    if (!amount) return ''
+  if (!amount) return ''
 
-    let slippageAmount = 0
-    if (isDeposit) slippageAmount = Math.ceil(+amount + ((slippage / 100) * +amount))
-    else slippageAmount = Math.floor(+amount - ((slippage / 100) * +amount))
-    return slippageAmount?.toString()
+  let slippageAmount = 0
+  if (isDeposit) slippageAmount = Math.ceil(+amount + (slippage / 100) * +amount)
```

Comment:
just checking is this UI units or native units; last time we had the UI break due to bonk values being really high e.g 500 mil overflowed, do we need to consider that or is this fine?